### PR TITLE
Fix BatchRenormalization clipping (fixes #127)

### LINKED ISSUE #127: BatchRenormalization uses Numpy code where it should use TF code
Looking at the code of [BatchRenormalization ](https://github.com/farizrahman4u/keras-contrib/blob/master/keras_contrib/layers/normalization.py), I think there is a serious issue there. This section, for example, inside the call() function:
```
# update r_max and d_max
t_val = K.get_value(self.t)
r_val = self.r_max_value / (1 + (self.r_max_value - 1) * np.exp(-t_val))
d_val = self.d_max_value / (1 + ((self.d_max_value / 1e-3) - 1) * np.exp(-(2 * t_val)))
t_val += float(self.t_delta)
```

The main issue is that the code uses Python/Numpy operations, rather than Keras/TensorFlow operations. Note that the layer call() function is executed only once when the model is created. It does *not* get called repeatedly for every batch. The call() function should create a TF/Theano computation graph rather than do the computation directly in Python/Numpy.

What's happening there is that it calls K.get_value(), which internally creates a TF session and runs it to get the float value of ```self.t```. Then, r_val and d_val and calculated in Python code and used in subsequent update ops. Since r_val and d_val are computed only once when the model is created, they'll have values of 1 and 0 respectively and will remain so. In other words, if I'm reading this correctly, this layer is operating as a regular batch normalization layer, not as a batch re-normalization layer. 

I didn't run the code to confirm so I could be missing something. If so, I'd appreciate being corrected.

### PR DESCRIPTION
**Pull request by stygstra**
Fixed the `r_max` and `d_max` clipping schedule. It was previously fixed at r_max=1 and d_max=0, making it essentially equivalent to regular batch normalization. See [my comment](https://github.com/farizrahman4u/keras-contrib/issues/127#issuecomment-325903950) in #127.

(Edit 2017-10-10: Removed parts that @titu1994 fixed in #151.)

### CODE REVIEW

File: keras_contrib/backend/tensorflow_backend.py [hunk 1]
```diff
@@ -158,3 +160,22 @@ def moments(x, axes, shift=None, keep_dims=False):
     ''' Wrapper over tensorflow backend call '''
 
     return tf.nn.moments(x, axes, shift=shift, keep_dims=keep_dims)
+
+
+def clip(x, min_value, max_value):
```
Comment (stygstra) [inline keras_contrib/backend/tensorflow_backend.py]: This was modified because the `K.clip` in Keras doesn't allow tensors for `min_value` or `max_value`.

**Comment by titu1994** [comment]
Hmm I corrected a few of these fixes in my PR. However, I didn't need any of these clip additions.

**Comment by stygstra** [comment]
@titu1994 I'll rebase onto master. The clipping-related changes address the issue noted in [this comment](https://github.com/farizrahman4u/keras-contrib/issues/127#issuecomment-325903950).

File: keras_contrib/backend/cntk_backend.py [hunk 2]
```diff
@@ -1,2 +1,10 @@
 from keras.backend import cntk_backend as KCN
 import cntk as C
+import numpy as np
+
+
+def clip(x, min_value, max_value):
+    if max_value is None:
```
Comment (ahundt) [inline keras_contrib/backend/cntk_backend.py]: if max_value defaults to inf, should min default to negative infinity?

Comment (stygstra) [reply keras_contrib/backend/cntk_backend.py]: You're correct. The Keras CNTK backend uses -inf as the default min_value. I did it this way because the TensorFlow backend *doesn't* default to -inf for min_value ([ref](https://github.com/fchollet/keras/blob/2.0.8/keras/backend/tensorflow_backend.py#L1472-L1489)), but failed to notice that the behavior of the CNTK backend differed.

**Comment by ahundt** [comment]
@stygstra Could you resolve the conflicts here?

File: keras_contrib/backend/cntk_backend.py (hunk 2 above)
Comment (stygstra) [reply keras_contrib/backend/cntk_backend.py]: I changed my mind and extended all backends to allow `None` for `min_value`. See notes in `tests/keras_contrib/backend/backend_test.py` for rationale and details.

**Comment by stygstra** [comment]
@ahundt Conflicts resolved. I changed the behavior of K.clip slightly since the previous iteration. See my comment in the review.

File: keras_contrib/backend/tensorflow_backend.py (hunk 1 above)
Comment (ahundt) [reply keras_contrib/backend/tensorflow_backend.py]: Documentation needs to be updated accordingly. Could these changes be valuable in keras itself?

File: keras_contrib/backend/cntk_backend.py (hunk 2 above)
Comment (ahundt) [reply keras_contrib/backend/cntk_backend.py]: Documentation needs to be updated accordingly. Could these changes be valuable in keras itself?

File: keras_contrib/backend/tensorflow_backend.py (hunk 1 above)
Comment (stygstra) [reply keras_contrib/backend/tensorflow_backend.py]: @ahundt Updated docstrings for all 3 backends. I think these changes could be useful in Keras for people writing new layers.
